### PR TITLE
Fixing issue where an unrecoverable failure of a task could be retried

### PR DESCRIPTION
**Pull request by dbrasseur-aneo**
This should fix an issue where if the worker is in a bad state (no dll, asked function doesn't exist), the task would still be retried.
Fix based on this flow : https://github.com/aneoconsulting/ArmoniK.Core/blob/main/Documentation/articles/tasks.md
The task will not be retried anymore if such exception arises

fix https://github.com/aneoconsulting/ArmoniK/issues/584

### CODE REVIEW

File: WorkerApi/ArmoniK.DevelopmentKit.WorkerApi/Services/ComputerService.cs [hunk 1]
```diff
@@ -155,8 +155,13 @@ public override async Task<Output> Process(ITaskHandler taskHandler)
       Logger.LogError(ex,
                       "WorkerAPIException failure while executing task");
 
-      throw new RpcException(new Status(StatusCode.Aborted,
-                                        ex.Message + Environment.NewLine + ex.StackTrace));
+      output = new Output
```
Comment (aneojgurhem) [inline WorkerApi/ArmoniK.DevelopmentKit.WorkerApi/Services/ComputerService.cs:158]: I wonder if a return here wouldn't be better ?

**Review by aneojgurhem — APPROVED** [review]

File: WorkerApi/ArmoniK.DevelopmentKit.WorkerApi/Services/ComputerService.cs (hunk 1 above)
Comment (dbrasseur-aneo) [reply WorkerApi/ArmoniK.DevelopmentKit.WorkerApi/Services/ComputerService.cs]: Done

File: WorkerApi/ArmoniK.DevelopmentKit.WorkerApi/AppsLoader.cs [hunk 2]
```diff
@@ -70,24 +70,28 @@ public class AppsLoader : IAppsLoader
...
     var localPathToAssemblyGridWorker = $"{Path.GetDirectoryName(localPathToAssembly)}/{ArmoniKDevelopmentKitServerApi}.dll";
 
-    assemblyGridWorker_ = UserAssemblyLoadContext.LoadFromAssemblyPath(localPathToAssemblyGridWorker);
-
-    if (assemblyGridWorker_ == null)
+    try
+    {
+      assemblyGridWorker_ = UserAssemblyLoadContext.LoadFromAssemblyPath(localPathToAssemblyGridWorker);
+    }
+    catch (Exception ex)
```
Comment (aneojgurhem) [inline WorkerApi/ArmoniK.DevelopmentKit.WorkerApi/AppsLoader.cs:91]: Vraiment toutes ?

**Review by aneojgurhem — APPROVED** [review]

File: WorkerApi/ArmoniK.DevelopmentKit.WorkerApi/AppsLoader.cs (hunk 2 above)
Comment (dbrasseur-aneo) [reply WorkerApi/ArmoniK.DevelopmentKit.WorkerApi/AppsLoader.cs:91]: https://learn.microsoft.com/en-us/dotnet/api/system.runtime.loader.assemblyloadcontext.loadfromassemblypath?view=net-6.0
Y'en a pas mal, ça ferait beaucoup de lignes pour pas grand chose

Comment (aneojgurhem) [reply WorkerApi/ArmoniK.DevelopmentKit.WorkerApi/AppsLoader.cs:91]: ok

**Pull request merged**